common/src/auth/ServiceConnectionStringParser.cs

```diff
@@ -0,0 +1,228 @@
+        /// The Service Client Hostname
+        /// </summary>
+        public string HostName
+        {
+            get { return _hostName; }
+            set { SetHostName(value); }
+        }
+
+        /// <summary>
+        /// The Service Access Policy Name
+        /// </summary>
+        public string SharedAccessKeyName { get; internal set; }
+
+        /// <summary>
+        /// The Service Shared Access Key for the specified
+        /// access policy
+        /// </summary>
+        public string SharedAccessKey { get; internal set; }
+
+        /// <summary>
+        /// The Service Shared Access Signature
+        /// </summary>
+        public string SharedAccessSignatureString { get; internal set; }
+
+        /// <summary>
+        /// The Service Name
+        /// </summary>
+        public string ServiceName
+        {
+            get { return _serviceName; }
+        }
+
+        internal ServiceConnectionString ToServiceConnectionString()
+        {
+            Validate();
+            return new ServiceConnectionString(this);
+        }
+
+        /// <summary>
+        /// Returns the Service Connection string
+        /// </summary>
+        /// <returns></returns>
+        public override string ToString()
+        {
+            Validate();
+
+            StringBuilder stringBuilder = new StringBuilder();
+            stringBuilder.AppendKeyValuePairIfNotEmpty(HostNamePropertyName, HostName);
+            stringBuilder.AppendKeyValuePairIfNotEmpty(SharedAccessKeyNamePropertyName, SharedAccessKeyName);
+            stringBuilder.AppendKeyValuePairIfNotEmpty(SharedAccessKeyPropertyName, SharedAccessKey);
+            stringBuilder.AppendKeyValuePairIfNotEmpty(SharedAccessSignaturePropertyName, SharedAccessSignatureString);
+            if (stringBuilder.Length > 0)
+            {
+                stringBuilder.Remove(stringBuilder.Length - 1, 1);
+            }
+
+            return stringBuilder.ToString();
+        }
+
+        protected virtual void Parse(string serviceConnectionString)
+        {
+            IDictionary<string, string> map = serviceConnectionString.ToDictionary(ValuePairDelimiter, ValuePairSeparator);
+
+            HostName = GetConnectionStringValue(map, HostNamePropertyName);
+            SharedAccessKeyName = GetConnectionStringOptionalValue(map, SharedAccessKeyNamePropertyName);
+            SharedAccessKey = GetConnectionStringOptionalValue(map, SharedAccessKeyPropertyName);
+            SharedAccessSignatureString = GetConnectionStringOptionalValue(map, SharedAccessSignaturePropertyName);
+            Validate();
+        }
+
+        protected virtual void Validate()
+        {
+            if (string.IsNullOrWhiteSpace(ServiceName))
+            {
+                throw new FormatException("Missing service name");
+            }
+
+            if (!string.IsNullOrWhiteSpace(SharedAccessKey))
+            {
+                Convert.FromBase64String(SharedAccessKey);
```

Comment:
Did you mean to get the return value? Are you just using it to validate? If so, I'd comment "Calling for validation".

Reply:
I am not sure if I understand your comment here! Here it is converting to SharedAccessKey base64 if not null, validate later

Reply:
We are using this as validation, if not base-64 an exception will be thrown. I have added a comment for the same.